src/prefect/results.py

```diff
@@ -209,7 +219,7 @@ def should_persist_result() -> bool:
     if flow_run_context is not None:
         return flow_run_context.persist_result
 
-    return PREFECT_RESULTS_PERSIST_BY_DEFAULT.value()
+    return get_default_persist_setting()
```

Comment:
it looks like this utility will return `True` if the flow run has persistence turned on, regardless of the task-specific setting; what am I missing?

Reply:
It will, but we check the new setting before calling this when setting up the task run context like this:
```python
if self.task.persist_result is not None:
    persist_result = self.task.persist_result
elif settings.tasks.default_persist_result is not None:
    persist_result = settings.tasks.default_persist_result
else:
    persist_result = should_persist_result()
```
Another implementation I considered was to include that check in this util, but the utility would need to know if a task was the one calling the function via a kwarg. This question makes me think that might be a better implementation so that all the determination logic is collocated and easier to find.

Reply:
The tricky part is that `should_persist_result` is the source of truth for all cases except when we're about to set up a `TaskRunContext`, which is where we need a small escape hatch. Let me see what I can do to clean this up.